src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1243,6 +1247,35 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                             break;
                         }
 
+                    case DeclarationModifiers.File:
+                        // 'file' is only a modifier if it is followed by an optional sequence of modifiers and then by a type keyword.
+                        {
+                            for (int i = 1; ; i++)
```

Comment:
>for (int i = 1; ; i++) [](http://example.com/codeflow?start=28&length=22)

Is this a perf issue if the source is a long sequence of `"file file file ... "`? Can we avoid unbounded look ahead without making the method unnecessarily complicated? For instance, create a `ResetPoint` when we see the first modifier, then consume `file` tokens as modifiers unconditionally, and before returning, if the last modifier was a `file` modifier and the next token is not a type, reset?

Reply:
Perhaps we could do a single scan before 'ParseModifiers'. We already have a flag 'forAccessors', perhaps we could also indicate whether we are parsing modifiers for a type.